addons/stock/migrations/13.0.1.1/pre-migration.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2020 Payam Yasaie <https://www.tashilgostar.com>
+# Copyright 2020 ForgeFlow <https://www.forgeflow.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+_column_copies = {
+    'stock_location': [
+        ('usage', None, None),
+    ],
+}
+
+_model_renames = [
+    ('report.stock.forecast', 'report.stock.quantity'),
+    ('stock.fixed.putaway.strat', 'stock.putaway.rule'),
+]
+
+_table_renames = [
+    ('stock_fixed_putaway_strat', 'stock_putaway_rule'),
```

Comment:
I think it is a bad idea to rename the table because:
 1. you have empty column location_in_id
 2. not all rows
Example:
 You have two locations with the same putaway so after migration, you have only one record in table stock_putaway_rule instead of two records

Reply:
and what do you propose instead?

Reply:
you can join stock_fixed_putaway_strat and stock_location by putway and insert in stock_putaway_rule

Reply:
done